db/commitlog/commitlog.cc

```diff
@@ -1303,6 +1385,36 @@ db::commitlog::segment_manager::segment_manager(config c)
     }
 }
 
+void db::commitlog::segment_manager::send(message m) {
+    _message_queue.push(std::move(m));
+}
+
+void db::commitlog::segment_manager::wakeup() {
+    if (_message_queue.empty()) {
+        _message_queue.signal();
+    }
+}
+
+future<> db::commitlog::segment_manager::message_loop() {
+    bool shutdown = false;
+
+    while (!shutdown || !_message_queue.empty()) {
+        auto next_wakeup = no_timeout;
+        auto now = message_clock::now();
+
+
+        co_await _message_queue.wait(next_wakeup).handle_exception([](auto ignored) {});
+
+        while (!_message_queue.empty()) {
+            auto m = _message_queue.pop();
+
+            std::visit(overloaded_functor {
+                [&](const shutdown_request&) { shutdown = true; },
+            }, m);
```

Comment:
IMO a condition variable + side channel for shutdown can be simpler (if you only have a single message type apart from shutdown_request).

Reply:
The idea is to try to serialize things, so multiple channels sort of defeat that idea... And even though we don't add too many messages here, I like having the option to.